pos_main.py

```diff
@@ -0,0 +1,162 @@
+                if done == 'y' or done == "Y":
+                    gluster_volume_commit_drive_swap(volname, old_brick, new_brick)
+                    break
+                else:
+                    continue
+            except GfCommandFailed as er:
+                        glogger.exception(f'replace brick failed for {old_brick} and {new_brick}')
+                        continue
+            else:
+                glogger.info(f'replace brick is done for {old_brick} and {new_brick}')
+
+
+def gf_migrate_bricks_using_swap_map(volname, bricks_migration_map):
+    """Swap bricks using migration map.
+
+    Ask user how the migration should happen.
+    Input:
+    volname: Name of the volume
+    bricks_migration_map: Physical migration map of old and new bricks.
+    """
+    while True:
+        try:
+            print("All together : 1")
+            print("One brick at a time : 0")
+            all = int(input("How do you want to swap device: (0 or 1) "))
+            if all not in [0, 1]:
+                print("Please enter correct option.")
+                continue
+            else:
+                break
+        except:
+            print("Please enter valid option")
+            glogger.debug('Invalid swap option')
+            ex = input("Want to exit? y/N")
+            if ex == "Y" or ex == "y":
+                exit(1)
+            else:
+                continue
+
+    if (all):
+        gf_swap_all_bricks(volname, bricks_migration_map)
+    else:
+        gf_swap_one_by_one_brick(volname, bricks_migration_map)
+
+
+def main():
+    """Main function to start complete plus one scaling process."""
+    while True:
+        try:
+            volname, filename, hostname = gf_get_volume_and_bricks()
+        except ValueError as er:
+            print(er)
+            glogger.exception(er)
+        else:
+            print(f'volname = {volname}')
+            print(f'filename = {filename}')
+            print(f'hostname = {hostname}')
+            break
+    try:
+        list_of_empty_drives = gf_get_new_bricks_list(filename)
+        print ("**************New Bricks*****************")
+        for br in list_of_empty_drives:
+            print(br)
+        print ("****************************************")
+    except GfCommandFailed as e:
+        print(e.message)
+        exit(1)
+
+    if not gluster_volume_is_ready_to_scale(volname, list_of_empty_drives):
+        print(f'EC volume, {volname}, can not be scaled. Volume not healthy or all the bricks are not UP')
+        exit(1)
+    else:
+        print(f'{volname} can be scaled')
+
+    print ("****************************************")
+    bricks_migration_map = gluster_create_brick_map_to_swap(
+        vol_name=volname, new_host=hostname,
+        list_of_empty_drives=list_of_empty_drives)
+
+    for old, new in bricks_migration_map:
+        print (f'old_brick={old} and new brick={new}')
+
+    gf_migrate_bricks_using_swap_map(volname, bricks_migration_map)
+
+main()
```

Comment:
Use as below to avoid accidental execution if this file is imported as library by other files,

```
if __name__ == "__main__":
    main()
```